test/context.c

```diff
@@ -66,6 +66,19 @@ static void test_bootslot_no_bootslot(void)
 	r_context_clean();
 }
 
+static void test_bootslot_custom_bootloader(void)
+{
+	r_context_conf()->configpath = g_strdup("test/test-custom.conf");
+	r_context_conf()->configmode = R_CONTEXT_CONFIG_MODE_REQUIRED;
+	r_context_conf()->mock.proc_cmdline = "quiet";
+	r_context_conf()->mock.custom_bootloader_output = "A";
```

Comment:
You might want to write something like

```
g_assert_true(g_setenv("CUSTOM_STATE_PATH", "test/custom-state-current", TRUE));
```

here and provide the file with some pre-set information. As an alternative, you could let your custom script just return a fixed bootname first of all.